chore: Add concurrency limitation to binary deployment

SamC-Apadmi opened the pull request:
Stop the merging of multiple PRs in quick succession from triggering lots of builds

mattmapadmi commented:
That `cancel-in-progress` seems like it could be dangerous if it's halfway through uploading one of the many artifacts? You'd end up with a duff build that would only work on certain platforms?

SamC-Apadmi commented:
Good point... It triggers a deployment to the snapshots on every merge into develop and I basically want to avoid having loads of them run at the same time

Commit:
chore: Add concurrency limitation to binary deployment

Stop the merging of multiple PRs in quick succession from triggering lots of builds
